src/components/Reports/StaffExpenseReport/SettingsDialog/SettingsDialog.tsx

```diff
@@ -0,0 +1,288 @@
+import React from 'react';
+import {
+  Box,
+  Button,
+  Checkbox,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormControlLabel,
+  FormGroup,
+  MenuItem,
+  TextField,
+  Typography,
+} from '@mui/material';
+import { Form, Formik } from 'formik';
+import { DateTime } from 'luxon';
+import { useSnackbar } from 'notistack';
+import { useTranslation } from 'react-i18next';
+import * as Yup from 'yup';
+import { CustomDateField } from 'src/components/common/DateTimePickers/CustomDateField';
+import { DateRange } from '../Helpers/StaffReportEnum';
+
+interface SettingsDialogProps {
+  isOpen: boolean;
+  selectedFilters?: Filters;
+  onClose: (filters?: Filters) => void;
+}
+
+export interface Filters {
+  selectedDateRange: DateRange | undefined;
+  startDate?: DateTime | null;
+  endDate?: DateTime | null;
+  // change to enum array when category data is provided
+  categories?: string[] | null;
+}
+
+const validationSchema = Yup.object({
+  selectedDateRange: Yup.mixed().nullable(),
+  startDate: Yup.mixed().nullable(),
+  endDate: Yup.mixed().nullable(),
+  categories: Yup.array().of(Yup.string()),
+});
+
+const calculateDateRange = (
+  range: DateRange,
+): { startDate: DateTime; endDate: DateTime } => {
+  const now = DateTime.now();
+
+  switch (range) {
+    case DateRange.WeekToDate:
+      return {
+        startDate: now.startOf('week'),
+        endDate: now.endOf('day'),
+      };
+    case DateRange.MonthToDate:
+      return {
+        startDate: now.startOf('month'),
+        endDate: now.endOf('day'),
+      };
+    case DateRange.YearToDate:
+      return {
+        startDate: now.startOf('year'),
+        endDate: now.endOf('day'),
+      };
+    // consider a default case if needed
+  }
+};
+
+export const SettingsDialog: React.FC<SettingsDialogProps> = ({
+  isOpen,
+  onClose,
+  selectedFilters,
+}) => {
+  const { t } = useTranslation();
+  const { enqueueSnackbar } = useSnackbar();
+
+  const [hasDateRangeError, setHasDateRangeError] = React.useState(false);
```

Comment:
You can import `useState` from `react`.

```suggestion
  const [hasDateRangeError, setHasDateRangeError] = useState(false);
```